answers/AbstractClassesAndMethods/8d32d57a-0fc1-4b3c-a85c-04a49a05b866/Animal.php

```diff
@@ -0,0 +1,35 @@
+<?php
+abstract class Animal
+{
+    private $type = "Dog";
+    private $age;
+	
+    public function __construct($type, $age)
+    {
+        $this->type = $type;
+        if ($this->isValid($age)) {
+            $this->age = $age;
+        }
```

Comment:
Same with this one. I think the question I'm trying to ask is, if the age is not valid, should we allow the creation of the object?

Reply:
In the implementation, if the value that you want to assign to age is not valid then this will not allow you to set a value to the `age` property but it will still create the object.

Reply:
Yeah, that is what I'm trying to avoid. Design wise, you shouldn't allow an object to be created  when it's invalid. E.g. there is no animal that has no age. I know this is a bit advance, but let me know if we should consider this design principle in our lesson.

Reply:
@cerilacanda can we throw an Exception in this lesson?